### PR TITLE
Argument validation + error handling in addTask

### PR DESCRIPTION
**Pull request by alongalky**

### CODE REVIEW

File: routes/api/tasks.js [hunk 1]
```diff
@@ -1,20 +1,38 @@
...
-  const params = {
-    command: req.body.command && req.body.command.join(' '),
-    output: req.body.output,
-    machine: req.body.machine,
-    key: req.key,
-    taskName: req.body.taskName,
-    tier: req.body.tier,
-    account: req.params.account_id
-  }
+  // Validation
```
Comment (karimodm) [inline routes/api/tasks.js]: You miss :account_id validation

File: test/routes/api-test.js [hunk 2]
```diff
@@ -56,15 +62,73 @@ describe('API', () => {
...
         .expect(200)
         .end((err, res) => {
-          sinon.assert.calledWithMatch(database.tasks.addTask, {account: '1234'})
+          sinon.assert.alwaysCalledWithMatch(database.tasks.addTask, {
+            command: 'hello world',
+            output: '/output',
+            machine: 'machina',
+            taskName: 'tasky',
+            tier: 'gargantuan',
+            account: '1234'
```
Comment (karimodm) [inline test/routes/api-test.js]: This should change to a valid value

File: test/routes/api-test.js [hunk 3]
```diff
@@ -56,15 +62,73 @@ describe('API', () => {
...
+          })
+
+          done(err)
+        })
+    })
+    it('returns 422 when command is not in array format', done => {
```
Comment (karimodm) [inline test/routes/api-test.js]: If you really want to test all cases, also test all parameters according to all the defined constraints added after resolving my first comment.

File: routes/api/tasks.js (hunk 1 above)
Comment (alongalky) [inline routes/api/tasks.js]: Fixed

File: routes/api/tasks.js [hunk 4]
```diff
@@ -1,20 +1,38 @@
...
+    }
+
+    const params = {
+      command: req.body.command && req.body.command.join(' '),
+      output: req.body.output,
+      machine: req.body.machine,
+      key: req.key,
+      taskName: req.body.taskName,
+      tier: req.body.tier,
+      account: req.params.account_id
```
Comment (alongalky) [inline routes/api/tasks.js]: Fixed

**Comment by alongalky** [comment]
Went over review, can you approve?

**Pull request merged**